ros/src/waypoint_follower/src/pure_pursuit_core.cpp

```diff
@@ -271,6 +288,43 @@ geometry_msgs::Twist PurePursuit::calcTwist(double curvature, double cmd_velocit
   return twist;
 }
 
+// edufford start
+// Search for closest waypt from current pose
+void PurePursuit::getClosestWaypoint() {
+  int path_size = static_cast<int>(current_waypoints_.getSize());
+  int closest_wp = path_size - 1;
+
+  // If waypoints are not given, do nothing
+  if (path_size == 0) {
+    closest_waypoint_idx_ = -1;
+    return;
+  }
+
+  // Initialize distance to first waypoint
+  double dist = getPlaneDistance(current_waypoints_.getWaypointPosition(0),
+                                 current_pose_.pose.position);
+
+  // Search for a closer waypoint
+  for (int i = 1; i < path_size; i++) {
+    double t_dist = getPlaneDistance(current_waypoints_.getWaypointPosition(i),
+                                     current_pose_.pose.position);
+    if (t_dist < dist) {
+      // Found a closer waypoint, store dist and keep searching
+      dist = t_dist;
+    } else {
+      // Distance is not decreasing, prev waypt was closest so stop searching
+      closest_wp = i - 1;
+      break;
+    }
+  }
+
+  // Store closest waypoint as output
+  closest_waypoint_idx_ = closest_wp;
+  //ROS_ERROR_STREAM("wp = " << closest_waypoint_idx_ << " dist = " << dist);
```

Comment:
Remove commented line

Reply:
I want to leave this one because it's nice to use for debug logging in case we need it later.

Reply:
Sounds good.